### ISSUE TITLE
Defensive melee style alerts despite having no defense-boosting potions

### ISSUE THREAD

**Issue by Ezeem** [AnkouOSRS/unpotted-reminder#4]
When training melee on the defensive style, checks for boosted defense stat and will constantly alert despite not having a defense-boosting potion in the inventory. This makes it very frustrating to use when on defensive, doing light slayer, etc.

There's a few potential solutions to this, but the most robust might be to include checking whether potions in the player's inventory actually boost a skill before warning for potion usage.

Mapping together the tracked skill to the potion that affects it and including that in the check here  https://github.com/AnkouOSRS/unpotted-reminder/blob/105dfa5ab0622cdc96e88d54addda7378d0e13c2/src/main/java/com/unpottedreminder/UnpottedReminderPlugin.java#L292-L293 or earlier on in the potions check here https://github.com/AnkouOSRS/unpotted-reminder/blob/105dfa5ab0622cdc96e88d54addda7378d0e13c2/src/main/java/com/unpottedreminder/UnpottedReminderPlugin.java#L275-L276
e.g.
```ts
POTIONS_BOOSTING_SKILL: Map<Integer, Array<Integer>>= { 
  Skill.DEFENCE: [
    ItemId.SUPER_COMBAT_POTION4, ItemId.BATTLEMAGE_POTION4
  ],
  Skill.STRENGTH: [
    ItemId.SUPER_COMBAT_POTION4, ItemId.SUPER_STRENGTH_POTION4
  ]
}
```

Biggest downside here would be that this ignores the case of a player forgetting a potion, but similarly the thrall helper has the option to not activate if you're not on the Arceeus spellbook.

**Comment by AnkouOSRS** [comment]
Actually I think I misunderstood your problem. On defensive it does check your boost for the defense skill, so even though it checks for attack/strength potions, to your point, if you are only boosting those and not boosting defense it probably does trigger when it shouldn't. I get what you're saying.

I'll look into fixing this.

**Comment by Ezeem** [comment]
Yeah that's exactly it, I appreciate you looking into it.

**Comment by AnkouOSRS** [comment]
This has been addressed in version 1.0.3. Now when using defensive melee it will check the boost for your strength skill instead of defense. Defensive ranged and magic will also be supported. I have submitted the PR so it will be updated on the plugin hub as soon as that is approved and merged
